File: lib/models/transcription.dart
class Transcription {
  final String id;
  final String sessionId;
  final String text;
  final DateTime timestamp;
  final String audioPath;

  Transcription({
    required this.id,
    this.sessionId = 'default_session',
    required this.text,
    required this.timestamp,
    required this.audioPath,
  });

  factory Transcription.fromJson(Map<String, dynamic> json) {
    return Transcription(
      id: json['id'],
      sessionId: json['sessionId'] ?? 'default_session',
      text: json['text'],
      timestamp: DateTime.parse(json['timestamp']),
      audioPath: json['audioPath'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'sessionId': sessionId,
      'text': text,
      'timestamp': timestamp.toIso8601String(),
      'audioPath': audioPath,
    };
  }
}

File: models/Transcription.ts
export interface Transcription {
  id: string;
  sessionId: string;
  text: string;
  timestamp: Date;
  audioPath: string;
}

export interface TranscriptionJSON {
  id: string;
  sessionId: string;
  text: string;
  timestamp: string;
  audioPath: string;
}

export const transcriptionToJSON = (transcription: Transcription): TranscriptionJSON => {
  return {
    id: transcription.id,
    sessionId: transcription.sessionId,
    text: transcription.text,
    timestamp: transcription.timestamp.toISOString(),
    audioPath: transcription.audioPath,
  };
};

export const transcriptionFromJSON = (json: TranscriptionJSON): Transcription => {
  const parsedDate = new Date(json.timestamp);
  if (isNaN(parsedDate.getTime())) {
    throw new Error(`Invalid timestamp: ${json.timestamp}`);
  }

  return {
    id: json.id,
    sessionId: json.sessionId ?? 'default_session',
    text: json.text,
    timestamp: parsedDate,
    audioPath: json.audioPath,
  };
};

export const createTranscription = (params: {
  id: string;
  sessionId?: string;
  text: string;
  timestamp: Date;
  audioPath: string;
}): Transcription => {
  return {
    id: params.id,
    sessionId: params.sessionId ?? 'default_session',
    text: params.text,
    timestamp: params.timestamp,
    audioPath: params.audioPath,
  };
};


